applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -30,6 +34,31 @@ class ResponseFunctionBase(object):
     solving of primal (and adjoint) analysis ...
     """
 
+    def __init___(model_part, response_settings):
+        self._model_part = None
+        self._response_settings = None
+
+    @property
+    def model_part(self):
+        if self._model_part is None:
+            raise RuntimeError("ResponseFunctionBase: model_part was not set by derived response function!")
+        return self._model_part
+
+    @model_part.setter
```

Comment:
hm these functions are quite non-standard
Have you taken a look at the mechanical-solver how e.g. the B&S is handled?

Reply:
I personally like the python properties, of course it would work also with `get_`  `set_` functions, essentially it is the same to me.